lib/index.js

```diff
@@ -3745,6 +3745,118 @@ _.wrapCallback = function (f) {
 };
 
 /**
+ * Takes an object or a constructor function and returns that object or
+ * constructor with streamified versions of its function properties.
+ * Passed constructors will also have their prototype functions
+ * streamified.  This is useful for wrapping many node style async
+ * functions at once, and for preserving those functions' context.
+ *
+ * @id streamifyAll
+ * @section Utils
+ * @name _.streamifyAll(source)
+ * @param {Object | Function} source - the function or object with
+ * node-style function properties.
+ * @api public
+ *
+ * var fs = _.streamifyAll(require('fs'));
+ *
+ * fs.readFileStream('example.txt').apply(function (data) {
+ *     // data is now the contents of example.txt
+ * });
+ */
+
+
+var isES5 = (function () {
+  'use strict';
+  return Function.prototype.bind && !this;
+}());
+
+function keys (obj) {
+    var keys = [];
+    for (var k in obj) {
+        if (hasOwn.call(obj, k)) {
+            keys.push(k);
+        }
+    }
+    return keys;
+}
+
+function isClass (fn) {
+    if (!(typeof fn === 'function' && fn.prototype)) { return false; }
+    var getKeys = isES5 ? Object.getOwnPropertyNames : keys;
+    var allKeys = getKeys(fn.prototype);
+    return allKeys.length > 0 && !(allKeys.length === 1 &&
+            allKeys[0] === 'constructor');
+}
+
+function inheritedKeys (obj) {
+    var allProps = {};
+    var curr = obj;
+    var handleProp = function (prop) {
+        allProps[prop] = true;
+    };
+    while (Object.getPrototypeOf(curr)) {
+        var props = Object.getOwnPropertyNames(curr);
+        props.forEach(handleProp);
+        curr = Object.getPrototypeOf(curr);
+    }
+    return keys(allProps);
+}
+
+function wrapWithContext (f) {
+    return function () {
+        var self = this;
+        var args = slice.call(arguments);
+        return _(function (push) {
+            var cb = function (err, x) {
+                if (err) {
+                    push(err);
+                }
+                else {
+                    push(null, x);
+                }
+                push(null, nil);
+            };
+            f.apply(self, args.concat([cb]));
+        });
+    };
+}
+
+function streamifyAll (inp, suffix) {
+    // will not streamify inherited functions in ES3
+    var getKeys = isES5 ? inheritedKeys : keys;
+    var allKeys = getKeys(inp);
+
+    for (var i = 0, len = allKeys.length; i < len; i++) {
+        var key = allKeys[i];
+        var val;
+        // will skip context aware getters
+        try { val = inp[key]; } catch (e) {}
+        if (val && typeof val === 'function' &&
+            !isClass(val) && !val.__isStreamified) {
```

Comment:
Let's use `__HighlandStreamifiedFunction__` (nouns instead of verbs & trailing `__`) to match the style of the other flags being used in the library (`__HighlandStream__`, `__HighlandStreamError`, etc).